patches/api/0440-SpawnerSearchEvent.patch

```diff
@@ -0,0 +1,131 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TreemanKing <67459602+TreemanKing@users.noreply.github.com>
+Date: Wed, 16 Aug 2023 11:59:08 +1000
+Subject: [PATCH] SpawnerSearchEvent
+
+This patch introduces a new event, SpawnerSearchEvent. This event is designed to be fired before a mob spawner attempts to search for nearby players to activate. It provides the ability to control and customize the activation logic of mob spawners, preventing them from searching and spawning mobs if desired
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/SpawnerSearchEvent.java b/src/main/java/io/papermc/paper/event/entity/SpawnerSearchEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..72b0ae8a84825cf273eba0827fb0551c1bda27e8
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/SpawnerSearchEvent.java
+@@ -0,0 +1,118 @@
++package io.papermc.paper.event.entity;
++
++import com.google.common.base.Preconditions;
++import org.bukkit.Location;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Range;
++import org.jetbrains.annotations.Unmodifiable;
++import java.util.Collection;
++import java.util.Collections;
++
++/**
++ * Represents an event that is fired before a mob spawner attempts to search
++ * for nearby players to activate. This event occurs before any of the spawner logic,
++ * allowing cancellation to prevent the spawner from searching and spawning any mobs.
++ */
++public class SpawnerSearchEvent extends Event {
++    @NotNull private final Location spawnerLocation;
++    @NotNull private final Collection<Player> activatingPlayers;
++    private int backoffTicks;
++    @NotNull private Result searchResult = Result.DEFAULT;
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++
++    @ApiStatus.Internal
++    public SpawnerSearchEvent(@NotNull Location spawnerLocation, @NotNull Collection<Player> activatingPlayers, @Range(from = 0, to = Integer.MAX_VALUE) int backoffTicks) {
++        this.spawnerLocation = Preconditions.checkNotNull(spawnerLocation, "Spawner Location can not be null");
++        this.activatingPlayers = Preconditions.checkNotNull(activatingPlayers, "Activating players list cannot be null");
++        this.backoffTicks = backoffTicks;
++    }
++
++    /**
++     * Retrieves the location of the mob spawner associated with the event
++     * @return The location of the spawner
++     */
++    @NotNull
++    public Location getSpawnerLocation() {
++        return spawnerLocation.clone();
++    }
++
++    /**
++     * Retrieves an immutable collection of players who are close enough to activate the spawner
++     * @return A collection of activating players
```

Comment:
Missing a new line between the description and the return/parameters, this also applies for getSpawnerLocation and setSearchResult